types/src/auction/bid.rs

```diff
@@ -75,6 +84,11 @@ impl Bid {
         self.release_era
     }
 
+    /// Returns a mutable reference to the delegators of the provided bid
+    pub fn delegators_mut(&mut self) -> &mut BTreeMap<PublicKey, Delegator> {
```

Comment:
Looking at call sites perhaps it'd be possible to expose a method that manipulates delegators rather than giving away mutable access?

Reply:
I certainly considered it, but the usage patterns at the call sites are different enough that this seemed to be the best choice to me.